### PR TITLE
AGENT-532: Add integration tests

### PR DESCRIPTION
**Pull request by andfasano**
The following patch proposes a minimal framework to allow writing integration tests for the agent-tui. A couple of sample tests have been added to show its usage

### CODE REVIEW

File: tools/agent_tui/app_test.go [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+				app.Start().
+					// Move to the node form
+					SelectItem("No").
+
+					// Wait for the node form, and insert an invalid ip
+					FocusItem("Rendezvous IP Address").
+					ScreenTypeText("127.0.0.1").
+
+					// Press "Check connectivity" button
+					SelectItem("Check connectivity").
```
Comment (rwsu) [inline tools/agent_tui/app_test.go:69]: Similarly, there is a CONNECTIVITYCHECK string const that could be used here.

**Review by rwsu — COMMENTED** [review]
It is a very nice test framework you have developed so far. Have you given some thoughts on how to test the scenario where the user clicks on the "Configure Networking" button? When it is clicked, the app suspends and calls nmutil: https://github.com/openshift/agent-installer-utils/blob/fa3676ffdee9578d4f5fe8a258b7b59243e0ab02/tools/agent_tui/net/nmtui.go#L13

File: hack/go-test.sh [hunk 2]
```diff
@@ -0,0 +1,17 @@
+#!/bin/bash
+
+SCRIPT_DIR="$(dirname "$(realpath "${BASH_SOURCE[0]}")")"
+
+if [ "$IS_CONTAINER" != "" ]; then
+	go test ./... "${@}"
+else
+	if ! podman image exists agent-installer-utils-gotools; then
```
[the remark this answers is not in the document]
Comment (andfasano) [reply hack/go-test.sh:8]: AFAIK there's no simple way (at least locally, without a CI) to do that. We could have some sort of version check, but probably given that images update are not very frequent it's easier to do it manually when required (or once in a while)

File: tools/agent_tui/app_test.go [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+import (
+	"testing"
+)
+
+func TestInitialScreen(t *testing.T) {
+	app := NewAppTester(t)
+	defer app.Stop()
+
+	app.Start().
+		WaitForScreenContent(
```
[the remark this answers is not in the document]
Comment (andfasano) [reply tools/agent_tui/app_test.go:12]: That was my initial approach (to linearize the screen in a single string), and was worried for the same reason. But then I realized that the simulation screen size is fixed, so we could rely on that. Anyhow, if considered useful, we could add a WaitForScreenContent version that works in the way you suggested

**Comment by andfasano** [comment]
> NMTUIRunner

That's a good point and will definitely require more discussion. Unfortunately there's nothing much we could, as long as the operations will be performed by an external program (which is, from our point of view, a "black box"). When the app is suspended the current simulation screen is detached and do not receive any feedback nor accept any input.
We already stated that we don't want to create a new implementation of `nmtui`, but without relying on `tview` for the UI it would not possible to test this specific part.

**Review by rwsu — COMMENTED** [review]
/lgtm

**Comment by celebdor** [comment]
/override ci/prow/gofmt
/override ci/prow/govet
/override /ci/prow/images

They pass locally
image

**Pull request merged**